PR: Automated Mechanism for Target Cell Handover Selection and Deactivation of Underutilized Cell(s)

mostafa-ashraf1709 opened the pull request:
- Added 5G support for handover with a mechanism for target cell selection.
- Implemented forwarding of SubReqData Action definitions, with event triggers to KPMIndicationMsg.
- Migrated to a minimum CPP version of 17.
- TODO: Implement handling of cell deactivation.
- Integrated reporting of the CurrentCell in KPM, along with IMSI-OldCell demultiplexing.


Fixes:
 [29](https://github.com/Orange-OpenSource/ns-O-RAN-flexric/issues/29), [15](https://github.com/Orange-OpenSource/ns-O-RAN-flexric/issues/15).

mostafa-ashraf1709 committed:
prepared for new KPM, considerd sinr, switched reducedParams flag.

mostafa-ashraf1709 committed:
Report CurrentCell in KPM, added IMSI-OldCell demultiplexing.

mostafa-ashraf1709 committed:
TODO: handle cell turn off.

AyaKamal10 committed:
Migtrate to min CPP version 17.

AyaKamal10 committed:
Forwarding SubReqData Action def, event trigger to KPMIndicationMsg.

mostafa-ashraf1709 committed:
Applied actionDef rules, on KPM IndcMsg.

MinaYonan123 commented:
@mostafa-ashraf1709  
please revise your commits into src/mmwave/model/mmwave-enb-net-device.cc

MinaYonan123 committed:
Fix bugs in parsing action defination to KPM indication

AbdoKhaled27 committed:
Fix get target cell for turn off

MinaYonan123 committed:
Fix bugs

MinaYonan123 committed:
Parse dlprbUsage to handle Action Defintion

kamilkociszewski committed:
New mechanism for delivering KPIs for GUI.

kamilkociszewski committed:
ns3 now calculate cell energy consumption by calculating time in every cell state and multiplying it by energy consumption for state.
GUI follows new method for EC calculations.
Additionally, GUI can run predefined scenarios.

Known bug for wrong UE number presented in GUI when user choose predefined scenario without use of run flags.

MinaYonan123 committed:
Merge pull request #17 from kamilkociszewski/KPM_reporting_upgdare2new_g++

Fix for GUI KPI traces, new cell energy consumption calculation

MinaYonan123 committed:
TODO: Fix HO control corner cases to improve reliability

kamilkociszewski committed:
clean scenarios from e2andlog flag

MinaYonan123 committed:
Add PerformE2RCHO function to handle RC HO control

MinaYonan123 committed:
Merge branch 'KPM_reporting_upgdare2new_g++' of https://github.com/MinaYonan123/mmwave-LENA-oran into KPM_reporting_upgdare2new_g++

MinaYonan123 committed:
Handle Action Definition Format 4 with PRB Usage Calculation
